cunumeric/array.py

```diff
@@ -2056,11 +2117,11 @@ def clip(self, min=None, max=None, out=None) -> ndarray:
                 category=RuntimeWarning,
             )
             if out is not None:
-                self.__array__().clip(min, max, out=out)
+                self.__array__().clip(args[0], args[1], out=out)
                 return convert_to_cunumeric_ndarray(out, share=True)
             else:
                 return convert_to_cunumeric_ndarray(
-                    self.__array__.clip(min, max)
+                    self.__array__().clip(args[0], args[1])
```

Comment:
missing parens seems like a bug, also above: is `out` only allows to be a *numpy* array? Currently is it passed directly to `ndarray.clip` so it must be

Reply:
The missing parens was a bug. And indeed this function is supposed to handle both `out=np.ndarray` and `out=cn.ndarray`, but the latter was not handled correctly. Fixed as of 022363bd9afe4e0cf6703085835c4244a0751717 (assuming `cn.ndarray__array__()` always returns a view over the same data as the cunumeric array, which is a question for another time).

While looking at this I also discovered https://github.com/nv-legate/cunumeric/issues/483.